### ISSUE TITLE
Change licence to MPL2

### ISSUE THREAD

**Issue by jobisoft** [jobisoft/TbSync#158]
@ChrisJAllan OK
@ldmpub OK
@ovari OK
@tijuca OK
@DzamoNorton OK
@AlessandroMenti OK
@Dagefoerde OK
@wisiemilljungdahl OK
@puran2 PENDING

I know we had this just a couple of month ago but I have to ask you again for your consent to switch the licence of TbSync.

People from Thunderbird have asked me, if I would like to let TbSync become part of Thunderbird Core and I would be very happy to see that happen. However, this only works if TbSync is licenced unter MPL2.

There is no schedule yet and there are probably a lot of things to be done, before TbSync becomes part of Thunderbird Core, but this licence change would be the first step.

Thanks for your support
John

**Comment by ovari** [comment]
Fine. All the best in your endeavors.

**Comment by DzamoNorton** [comment]
Congratulations, consent given.

**Comment by Dagefoerde** [comment]
You have my consent.

**Comment by AlessandroMenti** [comment]
I agree to the license change.

**Comment by ldmpub** [comment]
Very good news!

Congratulations and of course: I give my consent for license change.

**Comment by ChrisJAllan** [comment]
Sounds great!

**Comment by benbucksch** [comment]
Thanks so much!

This depends on #157

**Comment by jobisoft** [comment]
@benbucksch I will swap out the uuid lib next week.

**Comment by wisiemilljungdahl** [comment]
Great news!
I agree.

**Comment by jobisoft** [comment]
There is something else we need to get resolved: The EAS provider is currently bundled with TbSync and I have to separate the two. So we will need to have a "TbSync" repository and "EAS-4-TbSync" repository.

In the long run, the "TbSync" repository will probably cease to exists, because its code will be part of comm-central. 

So I renamed this repository to EAS-4-TbSync, keeping all your contributions data, but will remove any UI related stuff and strip it down to become a pure provider for TbSync. I also created a new dummy "TbSync" repository and I will upload just the UI stuff there, which eventually will be part of Thunderbird/comm-central.

How to handle your contributions and your contributer status to TbSync Core/UI? This is mostly translation stuff. Since the new "TbSync" repository will be a temporary repository, does that matter at all to you? How to handle that properly?

@benbucksch Is there a way in comm-central, to add and honor contributers?

**Comment by benbucksch** [comment]
@jobisoft:
> I renamed this repository to EAS-4-TbSync, keeping all your contributions data

I think it would be clearer, if you were to make a fork of this repository, and name the fork EAS-4-TbSync. Then, in the fork, you can remove all the TbSync code, and here in this repo, you can remove the EAS code. If you fork a github repo, it also forks the git repo. The copy retains all the contribution history, so even the forked EAS-4-TbSync would have all the relevant contribution history for the files that you keep.

> Is there a way in comm-central, to add and honor contributers?

I think there might be, if you manage to make a proper merge. It's going to be a lot of work, needs knowledge of hg etc.. It may work, but I'm not sure it's worth it.

Alternatively, you can just make a code comment somewhere, where you thank your contributors and describe on a high level what they did, and add their names to the Thunderbird contributors file.

**Comment by benbucksch** [comment]
John told me that GitHub doesn't let him fork his own repo, but you can still do GitHub "New repository", not initialize it with a README, and then push the old git repo there, or at least merge them.

Worse case, you merge the 2 repos manually:
```
git clone git@github.com/you/newrepo
cd newrepo
git remote add oldrepo git@github.com/you/oldrepo
git fetch oldrepo
git merge oldrepo/master
git commit
git push
```
or something like that (not tested).
This preserves the contributor history.

**Comment by jobisoft** [comment]
I will try that later, thanks for the hint. 

FYI: I wanted to remove the EAS provider from TbSync anyhow, even if TbSync will not make it into comm-central.

**Comment by WDenda** [comment]
Am 01.11.18 um 18:24 schrieb John Bieling:
> I wanted to remove the EAS provider from TbSync anyhow, even if TbSync 
> will not make it into comm-central.

Will this affect my thunderbird calendar and contact sync, and how do I 
find out? I'm using exchange.memotoo.com for calendar, contact, and 
tasks. My fear is that I have to switch back to evolution mail, because 
the future of thunderbird and it's addons seems to be not reliable.

Thanks!
Wolfgang

**Comment by jobisoft** [comment]
It worked using the import feature after creating a blank repository. So the old TbSync repository with all the issues is back and EAS-4-TbSync is a copy of TbSync containing all the contribution data as well.

Next I will remove the EAS provider from TbSync and since that removes @puran2 contribution I can make the license change for TbSync later today. 

After that, I start stripping down EAS-4-TbSync and wait for feddback from @puran2 to switch its licence as well.

**Comment by jobisoft** [comment]
@WDenda This is just a structural change, the EAS provider is being moved into its own Add-On, which needs to be installed alongside of TbSync. Similar to what [DAV-4-TbSync](https://addons.thunderbird.net/de/thunderbird/addon/dav-4-tbsync/) is doing now already.

**Comment by benbucksch** [comment]
@WDenda : You can continue to use the EAS provider as before, you just install it differently.
If anything, moving TbSync in Thunderbird ensures much better that it an d also its providers will continue to work despite changes in the addon infrastructure. This is great news for the future of TbSync.

**Comment by WDenda** [comment]
> You can continue to use the EAS provider as before, you just install it differently.
@jobisoft @benbucksch 
Thank you. Hopefully, this transition will do better on Mint, than it went with Lightning.

**Comment by jobisoft** [comment]
@WDenda I will do this in two steps. First I will release EAS-4-TbSync and a new version of TbSync, which still contains the EAS provider, but TbSync will inform you about the needed install of of EAS-4-TbSync. With the next-to-next update of TbSync I will remove EAS from TbSync (but still inform users about the change, if the user has EAS accounts but no EAS provider installed).

**Comment by jobisoft** [comment]
The EAS provider (including all contributions from @puran2) has been removed. Licence of the TbSync repository (including licence headers) has been changed to MPL2.

@puran2 You contributions are not lost! They are still valued and part of EAS-4-TbSync and will be kept, if you agree to the licence change there (which you can do by simple responding here).

**Comment by puran2** [comment]
Hi John,
I sent you 2 mail to accept the license change. I did not know that you expected this here.
I apologize.

Puran2

**Comment by jobisoft** [comment]
Hi @puran2,

thanks! 

Now lets wait and see how this is going. I close this issue, but if there are still questions about this, just ask :-)

Thanks for helping to make this possible,
John

**Comment by benbucksch** [comment]
Thanks, everybody, for this, and making integration of TbSync into Thunderbird possible!

**Comment by jobisoft** [comment]
@ldmpub 

I think the UI will not change much, but we have to wait and see. 

As soon as this gets approved and I have become familiar with the Thunderbird build environment, I will start to work on patches for comm-central. The main work will be to adjust the patches to pass review, so I think the overall review process will determine the date of the final landing. 

We have a schedule in mind, but I think it is to early to give out any numbers (@benbucksch ???).

**Comment by benbucksch** [comment]
> I think it is to early to give out any numbers (@benbucksch ???)

Yes :)

**Comment by ldmpub** [comment]
Thanks and good work 😂

**Comment by jobisoft** [comment]
@AlessandroMenti 
@ovari 

Could you fix the two translations marked with FIXME here?
https://github.com/jobisoft/EAS-4-TbSync/blob/master/install.rdf

Thanks!

**Comment by ovari** [comment]
Done at: https://github.com/jobisoft/EAS-4-TbSync/pull/2

Will the ATN hu page for `EAS-4-TbSync` be created and localized to hu too?

Thank you

**Comment by ovari** [comment]
Thanks. Image caption(s) would need translation too.

**Comment by jobisoft** [comment]
Thanks! And yes, I will try to match the ATN translations to the avail translations in the Add-On(s). I will reach out to you, once it has landed.

**Comment by jobisoft** [comment]
Today I pushed the EAS provider Add-On to ATN and as soon as it has been reviewed, I will release a new version of TbSync, which informs the user about the separation and that he must install the EAS provider to be able to continue to use ActiveSync.

@AlessandroMenti 
@ovari 

Could you translate the info message for the users? You can do that by a PR to the `tbSync.strings` files of the `0.7.16.9-(last-TbSync-with-EAS)` branch, or by simply posting the translation here. This is the english text:

```
TbSync is evolving!

TbSync started as a pure sync Add-On for ActiveSync (EAS) but has since turned into a central sync manager for multiple protocols (for example CalDAV/CardDAV).

Therefore, the next logical step is to move the ActiveSync provider from TbSync into a separate provider Add-On. The next version of TbSync will not include the ActiveSync provider anymore. To get this transition done without any hassle, I strongly suggest to install the separate ActiveSync provider now!

Do you want to open the Add-On page of the ActiveSync provider to install it with just a few clicks?
```
If there are suggestion to change the english text itself, post that here as well.

Thanks!

**Comment by benbucksch** [comment]
@jobisoft : I would suggest to open a new issue about the EAS split.

> new version of TbSync, which informs the user about the separation and that he must install the EAS provider to be able to continue to use ActiveSync.

If you want to make it nice, you could look at the configured accounts and install the plugin automatically for them (or make a dialog that offers to install it).

**Comment by tijuca** [comment]
>  I would suggest to open a new issue about the EAS split.

Yes please, this is now completely offtopic, please close this issue.

**Comment by jobisoft** [comment]
Ok. I will look at Ben's code and if I still need translations, I ask again in a new issue. 

No further replies in this issue than. It is closed.
